packages/fxa-settings/src/lib/oauth/hooks.tsx

```diff
@@ -117,16 +115,9 @@ async function constructOAuthCode(
  * @param oauthCode
  * @returns
  */
-function constructOAuthRedirectUrl(
-  oauthCode: {
-    code: string;
-    state: string;
-    redirect: string;
-  },
-  redirectUri: string
-) {
+function constructOAuthRedirectUrl(oauthCode: OAuthCode, redirectUri?: string) {
   // Update the state of the redirect URI
-  let constructedRedirectUri = new URL(redirectUri);
+  let constructedRedirectUri = new URL(redirectUri || oauthCode.redirect);
```

Comment:
Note: currently double checking this to make sure it's right. 

This is what it looked like we were doing after going through AMO in stage and prod and network requests, but it looks like we perform a [`delete response.redirect`](https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/app/scripts/models/auth_brokers/oauth-redirect.js#L217) and then set the redirect with [`updateSearchString`](https://github.com/mozilla/fxa/blob/main/packages/fxa-content-server/app/scripts/lib/url.js#L205-L206).

When running through this locally as described, `redirectUri` in that function is coming through as `http://localhost:8080/api/oauth` despite it not being a query param. This might be related to my notes for #16078...

> This SubPlat link initially hits content-server at localhost:3030. We store that value [here](https://github.com/mozilla/fxa/blob/f66607fa6557b240de28c0625995eda6c2afa3bd/packages/fxa-content-server/app/scripts/views/base.js#L406), which loads even before the index page does, but redirect_to in the URL gets set to localhost:3031.

Reply:
I figured out what the weirdness was here.

Instead of relying on this `redirect` from the response, content-server references the [client info config](https://github.com/mozilla/fxa/blob/30204fe0fba7b7ad18609c4f093328c5a1f223d5/packages/fxa-content-server/app/scripts/models/reliers/oauth.js#L280-L281) when `redirect_uri` isn't provided. I'd already accounted for this with sync mobile which doesn't give us a `redirect_uri`, but I thought this was a special case.

I pushed an update that changes a couple things around for `integration.data.redirectUri || integration.clientInfo.redirectUri`.

We might need a 1-pointer follow up here, it looks like [in content-server we always check](https://github.com/mozilla/fxa/blob/51461cd051d72d61a5683190f541895f3b3094b5/packages/fxa-content-server/app/scripts/models/reliers/oauth.js#L254-L255) that the `integration.data.redirectUri` matches something we have set in our config.